src/gt4py/next/otf/compilation/build_systems/cmake.py

```diff
@@ -9,16 +9,47 @@
 from __future__ import annotations
 
 import dataclasses
+import os
 import pathlib
 import subprocess
-from typing import Optional
 
+from gt4py._core import definitions as core_defs
 from gt4py.next import config
 from gt4py.next.otf import languages, stages
 from gt4py.next.otf.compilation import build_data, cache, common, compiler
 from gt4py.next.otf.compilation.build_systems import cmake_lists
 
 
+def get_device_arch() -> str | None:
+    if core_defs.CUPY_DEVICE_TYPE == core_defs.DeviceType.CUDA:
+        # use `cp` from core_defs to avoid trying to re-import cupy
+        return core_defs.cp.cuda.Device(0).compute_capability  # type: ignore[attr-defined]
+    elif core_defs.CUPY_DEVICE_TYPE == core_defs.DeviceType.ROCM:
+        # TODO(egparedes): Implement this properly, either parsing the output of `$ rocminfo`
+        # or using the HIP low level bindings.
+        # Check: https://rocm.docs.amd.com/projects/hip-python/en/latest/user_guide/1_usage.html
+        return "gfx90a"
+
+    return None
+
+
+def get_cmake_device_arch_option() -> str:
+    cmake_flag: str
```

Comment:
```suggestion
    cmake_flag = ""
```
and remove the `case None`?

Reply:
Done